packages/graphql/lib/src/core/query_manager.dart

```diff
@@ -244,7 +244,8 @@ class QueryManager {
 
     try {
       // execute the request through the provided link(s)
-      response = await link.request(request).first;
+      response =
+          await link.request(request).timeout(Duration(seconds: 5)).first;
```

Comment:
I missed the question

>Please comment on the timeout time of 5 seconds, as we have not been able to determine whether it is appropriate.

Can be made configurable from the user when create the client?

Reply:
#1447 to be able to configure it.